grpc.go

```diff
@@ -768,10 +770,16 @@ func (a *agentGRPC) WaitProcess(ctx context.Context, req *pb.WaitProcessRequest)
 		return &pb.WaitProcessResponse{}, err
 	}
 
+	proc.exitCodeCh.RLock()
+
 	defer func() {
+		proc.exitCodeCh.RLock()
+	}()
+
+	defer proc.Do(func() {
 		proc.closePostExitFDs()
```

Comment:
Please also reset all process FDs in `closePostExitFDs` so that they are not closed multiple times.

Reply:
Yes,  it has already been enclosed in this defer block.